File: static/archive/Competitive-Programming/SPOJ - CHOCOLA.cpp
#include <bits/stdc++.h>
using namespace std;
int N, M, T;
long long Le, Ri, Ans, Arr[1001], Arr2[1001];
int main() {
  cin >> T;
  for (int t = 1; t <= T; ++t) {
    cin >> N >> M;
    Ans = 0;
    N--;
    M--;
    Le = 1;
    Ri = 1;
    for (int i = 1; i <= N; ++i)
      cin >> Arr[i];
    for (int i = 1; i <= M; ++i)
      cin >> Arr2[i];
    sort(Arr + 1, Arr + N + 1);
    sort(Arr2 + 1, Arr2 + M + 1);
    while (N > 0 && M > 0) {
      if (Arr[N] > Arr2[M]) {
        Ans += Arr[N] * Le;
        Ri++;
        N--;
      } else {
        Ans += Arr2[M] * Ri;
        Le++;
        M--;
      }
    }
    while (N > 0)
      Ans += Arr[N] * Le, N--;
    while (M > 0)
      Ans += Arr2[M] * Ri, M--;
    cout << Ans << endl;
  }
}

File: static/archive/Competitive-Programming/SPOJ - CHOCOLA.py
tc=int(input())
for t in range(tc):
    trash=input()
    n,m=input().split()
    n=int(n)-1
    m=int(m)-1
    arr1=[]
    arr2=[]
    for i in range(n):
        x=int(input())
        arr1.append(x)
    for i in range(m):
        x=int(input())
        arr2.append(x)
    arr1.sort()
    arr2.sort()
    ans=0
    le=1
    ri=1
    while n>0 and m>0:
        if arr1[n-1]>arr2[m-1]:
            ans+=arr1[n-1]*le
            ri+=1
            n-=1
        else:
            ans+=arr2[m-1]*ri
            le+=1
            m-=1
    while n>0:
        ans+=arr1[n-1]*le
        n-=1
    while m>0:
        ans+=arr2[m-1]*ri
        m-=1
    print(ans)